blockchain/blockdao.go

```diff
@@ -35,6 +39,8 @@ const (
 	receiptsNS                       = "rpt"
 
 	hashOffset = 12
+
+	maxCacheSize = 10000
```

Comment:
This can be a config field? There will be three cases:
- nodes that will not provide API service, cache is unneeded
- nodes with limited memory + provide API service, set maxCacheSize to something the memory can afford
- nodes need realtime data (e.g., for debugging), set maxCacheSize to 0 to disable it.